addons/website/models/website.py

```diff
@@ -849,13 +849,17 @@ def _handle_views_and_pages(views):
             if dependency_records:
                 model_name = self.env['ir.model']._display_name_for([model])[0]['display_name']
                 dependencies.setdefault(model_name, [])
-                dependencies[model_name] += [{
-                    'field_name': Model.fields_get()[column]['string'],
-                    'record_name': rec.name,
-                    'link': 'website_url' in rec and rec.website_url or f'/web#id={rec.id}&view_type=form&model={model}',
-                    'model_name': model_name,
-                } for rec in dependency_records]
-
+                for rec in dependency_records:
+                    if hasattr(rec, 'name'):
+                        record_name = rec.name
+                    else:
+                        record_name = f'{model_name} {rec.id}'
+                    dependencies[model_name] += [{
+                        'field_name': Model.fields_get()[column]['string'],
+                        'record_name': record_name,
+                        'link': 'website_url' in rec and rec.website_url or f'/web#id={rec.id}&view_type=form&model={model}',
+                        'model_name': model_name,
+                    }]
```

Comment:
Keep the initial `for` loop format, on top of being easier to read, it will (and should) be a one line fix.
> The first version of this fix used the getattr() [..] leads to security issue.

What do you mean by security issue? If you mean that `getattr` triggers a ci security, it doesn't mean that you should not use it. (You could bypass that with :point_down: but there is no point)
```py
'ok' if 'name' in self else 'ko'
```

`getattr` is exactly what you want instead of
```py
if hasattr(rec, 'name'):
    record_name = rec.name
else:
    record_name = f'{model_name} {rec.id}'
```
->
```py
record_name = getattr(rec, 'name', f'{model_name} {rec.id}')
```